docs/ddp.md

```diff
@@ -0,0 +1,243 @@
+# How to do DistributedDataParallel
```

Comment:
nit: Should we use backticks around `DistributedDataParallel` to be consistent with the other docs? e.g. https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#comparison-between-dataparallel-and-distributeddataparallel